packages/mui-material-next/src/ButtonGroup/ButtonGroup.tsx

```diff
@@ -197,7 +216,19 @@ const ButtonGroupRoot = styled('div', {
   },
 }));
 
-const ButtonGroup = React.forwardRef(function ButtonGroup(inProps, ref) {
+/**
+ *
+ * Demos:
+ *
+ * - [Button Group](https://mui.com/material-ui/react-button-group/)
+ *
+ * API:
+ *
+ * - [ButtonGroup API](https://mui.com/material-ui/api/button-group/)
+ */
+const ButtonGroup = React.forwardRef(function ButtonGroup<
```

Comment:
It should be cast to `as OverridableComponent<ButtonGroupTypeMap>` at the end ~ e.g. https://github.com/mui/material-ui/blob/7ebc41bf6c735953e227715d82b6114e2b84d3b1/packages/mui-material-next/src/Chip/Chip.tsx#L449